packages/manager/apps/pci-instances/src/data/hooks/instances/useInstances.ts

```diff
@@ -0,0 +1,199 @@
+  { pages }: InfiniteData<TInstanceDto[], number>,
+  limit: number,
+): TInstance[] =>
+  pages
+    .flatMap((page) => (page.length > limit ? page.slice(0, limit) : page))
+    .map((instanceDto) => ({
+      ...instanceDto,
+      status: getInstanceStatus(instanceDto.status),
+      addresses: instanceDto.addresses.reduce((acc, { type, ...rest }) => {
+        const foundAddresses = acc.get(type);
+        const ipAlreadyExists = !!foundAddresses?.find(
+          ({ ip }) => ip === rest.ip,
+        );
+        if (foundAddresses) {
+          if (ipAlreadyExists) return acc.set(type, [...foundAddresses]);
+          return acc.set(type, [...foundAddresses, rest]);
+        }
+        return acc.set(type, [rest]);
+      }, new Map<TAddressType, TAddress[]>()),
+    }));
+
+export const useInstances = (
+  projectId: string,
+  { limit, sort, sortOrder, filters }: TUseInstancesQueryParams,
+) => {
+  const queryClient = useQueryClient();
+  const filtersQueryKey = useMemo(
+    () =>
+      filters?.length > 0
+        ? [
+            'filter',
+            filters[0].label,
+            filters[0].comparator,
+            filters[0].value as string,
+          ]
+        : [],
+    [filters],
+  );
+  const queryKey = useMemo(
+    () =>
+      instancesQueryKey(projectId, [
+        'list',
+        'sort',
+        sort,
+        sortOrder,
+        ...filtersQueryKey,
+      ]),
+    [filtersQueryKey, projectId, sort, sortOrder],
+  );
+
+  const refresh = useCallback(() => {
+    queryClient.removeQueries({
+      predicate: (query) =>
+        query.queryKey.includes('list') &&
+        query.queryKey !==
+          instancesQueryKey(projectId, ['list', 'sort', 'name', 'asc']),
+    });
+  }, [projectId, queryClient]);
+
+  useEffect(() => {
+    const queryData = queryClient.getQueryData<InfiniteData<TInstance[]>>(
+      queryKey,
+    );
+    if (queryData?.pageParams && queryData.pageParams.length > 1) {
+      queryClient.resetQueries({
+        queryKey,
+        exact: true,
+      });
+    }
+  }, [projectId, sort, sortOrder, queryClient, queryKey, filters.length]);
+
+  const { data, ...rest } = useInfiniteQuery({
+    queryKey,
+    retry: false,
+    initialPageParam: 0,
+    getNextPageParam: (lastPage, _allPages, lastPageParam) =>
+      lastPage.length > limit ? lastPageParam + 1 : null,
+    queryFn: ({ pageParam }) =>
+      getInstances(projectId, {
+        limit,
+        sort,
+        sortOrder,
+        offset: pageParam * limit,
+        ...(filters.length > 0 && { searchField: filters[0].label }),
+        ...(filters.length > 0 && { searchValue: filters[0].value as string }),
+      }),
+    select: useCallback(
+      (rawData: InfiniteData<TInstanceDto[], number>) =>
+        instancesSelector(rawData, limit),
+      [limit],
+    ),
+    placeholderData: keepPreviousData,
+  });
+
+  return {
+    data,
+    hasInconsistency: getInconsistency(data),
+    refresh,
+    ...rest,
+  };
```

Comment:
@anooparveti 
As I said in my previous comment (above), we can't use these hooks because the `/instances` endpoint is not using Iceberg, and is using specific filtering / sorting / pagination behavior.
Furthermore, we need extra parameters like `select` or `placeholderData` that are not implemented by these hooks

Reply:
@fredericvilcot, we will add the capabilities like client-side sorting/searching/filtering on `useResourcesV6` hook.
For `placeholderdata` and `select` you can request them on the `manager-react-components` Github project.

Reply:
@anooparveti 
I will request extra parameters, thanks.
For the sorting/searching/filtering, in this case, we are using server-side logic but on a v6 endpoint not handled by Iceberg so I don't think it could be added to the `useResourcesV6` hook.
We will need a custom hook for that in the µapp to handle this specific use-case.